library/Zend/Db/Adapter/Pdo/Mysql.php

```diff
@@ -200,7 +200,7 @@ public function describeTable($tableName, $schemaName = null)
                 $row[$type] = 'float';
                 $precision = $matches[1];
                 $scale = $matches[2];
-            } else if (preg_match('/^((?:big|medium|small|tiny)?int)\((\d+)\)/', $row[$type], $matches)) {
+            } else if (preg_match('/^((?:big|medium|small|tiny)?int)(\((\d+)\))?/', $row[$type], $matches)) {
```

Comment:
use /^((?:big|medium|small|tiny)?int)(\(?:(\d+)\))?/
It will not break the group name order